src/main.rs

```diff
@@ -156,21 +158,44 @@ impl Fixer {
     /// subsequently query. Return a description of the failing operation on
     /// error.
     fn build_file_info(file_name: &str) -> Result<FileInfo, String> {
-        // Get the debug session from file.
-        let msg = |op| format!("Unable to {} file {}", op, file_name);
-        let data = fs::read(file_name).map_err(|_| msg("read"))?;
+        let msg = |op: &str| format!("Unable to {} `{}`", op, file_name);
+
+        // Read the file.
+        let mut data = fs::read(file_name).map_err(|_| msg("read"))?;
+
+        // On some platforms we have to get the debug info from another file.
+        // Get the name of that file, if there is one.
+        let file_name2 = match Object::peek(&data) {
+            FileFormat::Pe => {
+                let pe_object = Object::parse(&data).map_err(|_| msg("parse"))?;
+                if let Object::Pe(pe) = pe_object {
+                    // PE files should contain a pointer to a PDB file.
+                    let pdb_file_name = pe.debug_file_name().ok_or_else(|| msg("find PDB for"))?;
+                    Some(pdb_file_name.to_string())
+                } else {
+                    panic!(); // Impossible: peek() said it was a PE object.
+                }
+            }
+            _ => None,
+        };
+        if let Some(file_name2) = file_name2 {
+            data = fs::read(&file_name2)
+                .map_err(|_| msg(&format!("read debug info file `{}` for", file_name2)))?;
+        }
+
+        // Get the debug session from the file data.
         let object = Object::parse(&data).map_err(|_| msg("parse"))?;
         let debug_session = object
             .debug_session()
             .map_err(|_| msg("read debug info from"))?;
 
-        // Build the `FileInfo` from the debug session.
+        // Build the `FileInfo` from the debug session. `tests/README.md` has an
+        // explanation of the commented-out `eprintln!` statements.
         let mut interner = Interner::default();
-        let func_infos = debug_session
+        let mut func_infos: Vec<_> = debug_session
             .functions()
             .filter_map(|function| {
                 let function = function.ok()?;
-                // This eprintln! is useful when creating tests.
```

Comment:
Why did you remove this line?

Reply:
I moved the comment to the top of the loop, to avoid the comment duplication. And it now points to the `tests/README.md` file.